File: java/src/main/java/com/experian/dvs/client/address/Confidence.java
package com.experian.dvs.client.address;

import java.util.HashMap;
import java.util.Map;

public enum Confidence {
    UNKNOWN("Unknown"),
    VERIFIED_MATCH("Verified match"),
    MULTIPLE_MATCHES("Multiple matches"),
    TOO_MANY_MATCHES("Too many matches"),
    INTERACTION_REQUIRED("Interaction required"),
    PREMISES_PARTIAL("Premises partial"),
    STREET_PARTIAL("Street partial"),
    VERIFIED_PLACE("Verified place"),
    VERIFIED_STREET("Verified street"),
    INCOMPLETE_ADDRESS("Incomplete address"),
    INSUFFICIENT_SEARCH_TERMS("Insufficient search terms"),
    NO_MATCHES("No matches")
    ;

    private final String name;
    //Map of name to Confidence
    private static final Map<String, Confidence> nameToConfidence = new HashMap<>();
    static {
        for (Confidence confidence : Confidence.values()) {
            nameToConfidence.put(confidence.name, confidence);
        }
    }

    Confidence(final String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static Confidence fromName(final String name) {
        return nameToConfidence.get(name);
    }

}

File: typescript/src/address/confidence.ts
export enum Confidence {
    Unknown = "Unknown",
    VerifiedMatch = "Verified match",
    MultipleMatches = "Multiple matches",
    TooManyMatches = "Too many matches",
    InteractionRequired = "Interaction required",
    PremisesPartial = "Premises partial",
    StreetPartial = "Street partial",
    VerifiedPlace = "Verified place",
    VerifiedStreet = "Verified street",
    IncompleteAddress = "Incomplete address",
    InsufficientSearchTerms = "Insufficient search terms",
    NoMatches = "No matches"
}

export function lookupConfidence(str?: string): Confidence {
    if (str) {
        return Object.keys(Confidence).find(key => Confidence[key as keyof typeof Confidence] === str) as Confidence | Confidence.Unknown;    
    } else {
        return Confidence.Unknown;
    }
}
